src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,30 +1,68 @@
-export const TodoFilter = () => (
-  <form className="field has-addons">
-    <p className="control">
-      <span className="select">
-        <select data-cy="statusSelect">
-          <option value="all">All</option>
-          <option value="active">Active</option>
-          <option value="completed">Completed</option>
-        </select>
-      </span>
-    </p>
+import { Dispatch, SetStateAction } from 'react';
+import { StatusFilter } from '../../modules/StatusFilter';
 
-    <p className="control is-expanded has-icons-left has-icons-right">
-      <input
-        data-cy="searchInput"
-        type="text"
-        className="input"
-        placeholder="Search..."
-      />
-      <span className="icon is-left">
-        <i className="fas fa-magnifying-glass" />
-      </span>
+type Props = {
+  status: StatusFilter;
+  query: string;
+  setStatus: Dispatch<SetStateAction<StatusFilter>>;
+  setQuery: Dispatch<SetStateAction<string>>;
+};
 
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button data-cy="clearSearchButton" type="button" className="delete" />
-      </span>
-    </p>
-  </form>
-);
+export const TodoFilter: React.FC<Props> = ({
+  status,
+  query,
+  setStatus,
+  setQuery,
+}) => {
+  const STATUS_OPTIONS: Record<StatusFilter, string> = {
+    all: 'All',
+    completed: 'Completed',
+    active: 'Active',
+  };
+
+  return (
+    <form className="field has-addons">
+      <p className="control">
+        <span className="select">
+          <select
+            data-cy="statusSelect"
+            value={status}
+            onChange={event => setStatus(event.target.value as StatusFilter)}
+          >
+            {Object.entries(STATUS_OPTIONS).map(([optionStatus, text]) => (
+              <option key={optionStatus} value={optionStatus}>
+                {text}
+              </option>
+            ))}
+          </select>
+        </span>
+      </p>
+
+      <p className="control is-expanded has-icons-left has-icons-right">
+        <input
+          data-cy="searchInput"
+          type="text"
+          className="input"
+          placeholder="Search..."
+          value={query}
+          onChange={event => setQuery(event.target.value.trimStart())}
```

Comment:
You're using `trimStart()` method in `setQuery` function. It will remove the whitespace from the start of the search query. However, it might not be what you want in all cases. If a user wants to search for a phrase that begins with a space, this will prevent them from doing so. Consider removing `trimStart()` or replace it with `trim()` to remove spaces from both ends of the string if leading and trailing spaces are not important for your search functionality.